astropy/config/configuration.py

```diff
@@ -360,9 +359,11 @@ def set(self, value):
                 f" {e.args[0]}"
             )
 
+        # store value on the ConfigObj instance...
         sec = get_config(self.module, rootname=self.rootname)
-
         sec[self.name] = value
+        # and on the descriptor
+        self.value = value
```

Comment:
While you are at it, I think inside `set_temp` below, `self.set(value)` should really be inside the `try/finally` (if it raises an exception, the value should be unset).

Reply:
done